### PR TITLE
phpunit update

### PR DESCRIPTION
**Pull request by kokspflanze**
- PHPUnit_Framework_TestCase to PHPUnit\Framework\TestCase
- changed setExpectedException method to expectException, expectExceptionMessage
- fix "This test did not perform any assertions"
- @requires PHP 5.4 removed
- removed old PHPUnit_Runner_Version case
- changed phpunit 4 to 5.7/6 in composer
- added composer.lock
- removed composer.lock in gitignore
- travis process update
- added php 7.1 in travis

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,60 +2,78 @@ sudo: false
...
-  - if [[ $DEPLOY_DOCS == "true" && "$TRAVIS_TEST_RESULT" == "0" ]]; then wget -O theme-installer.sh "https://raw.githubusercontent.com/zendframework/zf-mkdoc-theme/master/theme-installer.sh" ; chmod 755 theme-installer.sh ; ./theme-installer.sh ; fi
-
-after_success:
-  - if [[ $DEPLOY_DOCS == "true" ]]; then echo "Preparing to build and deploy documentation" ; ./zf-mkdoc-theme/deploy.sh ; echo "Completed deploying documentation" ; fi
 
 after_script:
   - if [[ $TEST_COVERAGE == 'true' ]]; then composer upload-coverage ; fi
+
+notifications:
+  email: false
```
Comment (webimpress) [inline .travis.yml]: Please add one empty line at the end of the file

File: test/ArrayUtilsTest.php [hunk 2]
```diff
@@ -513,7 +514,7 @@ public function testFiltersArray($data, $callback, $flag, $result)
     }
 
     /**
-     * @expectedException \Zend\Stdlib\Exception\InvalidArgumentException
+     * @expectedException InvalidArgumentException
```
Comment (webimpress) [inline test/ArrayUtilsTest.php]: Just for consistency I would change it to method call instead of PHPDoc tag.

File: .travis.yml (hunk 1 above)
Comment (kokspflanze) [inline .travis.yml]: added, didnt saw it in the template, i didnt checkout the repo

File: test/ArrayUtilsTest.php (hunk 2 above)
Comment (kokspflanze) [inline test/ArrayUtilsTest.php]: changed

**Comment by froschdesign** [comment]
@kokspflanze 
Thanks! 👍

**Pull request merged**